modules/shortcodes/instagram.php

```diff
@@ -227,16 +309,15 @@ function jetpack_shortcode_instagram( $atts ) {
 		return '';
 	}
 
+	$atts = jetpack_instagram_get_whitelisted_parameters( $atts['url'], $atts );
+
+	if ( empty( $atts['url'] ) ) {
+		return '';
+	}
+
 	if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
 		$url_pattern = '#http(s?)://(www\.)?instagr(\.am|am\.com)/p/([^/?]+)#i';
 		preg_match( $url_pattern, $atts['url'], $matches );
-		if ( ! $matches ) {
-			return sprintf(
-				'<a href="%1$s" class="amp-wp-embed-fallback">%1$s</a>',
-				esc_url( $atts['url'] )
-			);
-		}
-
```

Comment:
I don't see at a glance why we're removing this :sweat_smile:

Reply:
I removed it since we're already checking for matches in `jetpack_instagram_get_allowed_parameters`, so by this point we know there is going to be a match. I'm happy to add it back though, if you think we'd better be safe.

Reply:
`jetpack_instagram_get_allowed_parameters` ensures that the URL begins with "https://www.instagram.com", but it may not have the path part as "/p/....".

Also the original code would be less odd-looking if it had been doing `if ( ! preg_match( ... ) ) {` rather than testing `$matches`.

Reply:
True. Do you think an approach like this would be better? 
https://github.com/Automattic/jetpack/pull/17700#discussion_r517516721

Reply:
Added it back in 91defe091a302887d3140118c3ac77eb9c08f087